sockeye/data_io.py

```diff
@@ -516,27 +532,28 @@ def prepare_data(source: str, target: str,
     data_statistics.log()
 
     data_loader = RawParallelDatasetLoader(buckets=buckets,
-                                           eos_id=vocab_target[C.EOS_SYMBOL],
+                                           eos_id=target_vocab[C.EOS_SYMBOL],
                                            pad_id=C.PAD_ID)
 
     # 3. convert each shard to serialized ndarrays
-    for shard_idx, (shard_source, shard_target, shard_stats) in enumerate(shards):
-        source_sentences = SequenceReader(shard_source, vocab=None)
+    for shard_idx, (shard_sources, shard_target, shard_stats) in enumerate(shards):
+        sources_sentences = [SequenceReader(s, vocab=None) for s in shard_sources]
         target_sentences = SequenceReader(shard_target, vocab=None)
-        dataset = data_loader.load(source_sentences, target_sentences, shard_stats.num_sents_per_bucket)
+        dataset = data_loader.load(sources_sentences, target_sentences, shard_stats.num_sents_per_bucket)
         shard_fname = os.path.join(output_prefix, C.SHARD_NAME % shard_idx)
         shard_stats.log()
         logger.info("Writing '%s'", shard_fname)
         dataset.save(shard_fname)
 
         if not keep_tmp_shard_files:
-            os.remove(shard_source)
+            for file in shard_sources:
```

Comment:
nitpicky: would it make sense to have another variable name other than `file`? (was a builtin in python 2 and get's still special highlights, e.g. by github)

Reply:
renamed to f